applications/DEMApplication/tests/random_variable_tests_files/random_variable.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+import random
+import matplotlib.pyplot as plt
+import KratosMultiphysics
+
+
+def GaussianPDF(x, mu=0, sigma=1):
+	return 1/(sigma * np.sqrt(2 * np.pi)) * np.exp( - (x - mu)**2 / (2 * sigma**2))
+
+class RandomVariable:
+	def __init__(self, parameters):
+		pass
+
+	def Sample(self,  n_to_pick=1):
+		return 0.0
+
+class PiecewiseLinearRV(RandomVariable):
+	def __init__(self, parameters):
+		if not isinstance(parameters, KratosMultiphysics.Parameters):
+			raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+		self.boundaries = np.array(parameters['breakpoints'].GetVector(), dtype=np.float32)
+		self.heights = np.array(parameters['values'].GetVector(), dtype=np.float32)
+		self.ranges = np.array([(self.boundaries[i+1] - self.boundaries[i] for i in range(len(self.boundaries)))])
+		self.areas = np.zeros(len(self.heights) - 1)
+		self.trapezoid_indices = range(len(self.areas))
+		self.Normalize()
+		super().__init__(parameters)
+
+	def InterpolateHeight(self, x):
+		if x < self.boundaries[0] or x > self.boundaries[-1]:
+			raise Exception('x =' + str(x) + 'is out of bounds (' + str([self.boundaries[0], self.boundaries[-1]]) + ')')
+
+		for i, x_boundary in enumerate(self.boundaries):
```

Comment:
sorry, the comment was meant to be here

Reply:
Immediately after (e.g., `b1 = self.heights[i - 1]`)